src/Time/Units.hs

```diff
@@ -75,13 +78,33 @@ type Day         = Time DayUnit
 type Week        = Time WeekUnit
 type Fortnight   = Time FortnightUnit
 
+-- | Type family for prettier 'show' of time units.
+type family ShowUnit (unit :: Rat) :: Symbol
+
+type instance ShowUnit (1 :% 1)          = "s"   -- second unit
+type instance ShowUnit (1 :% 1000)       = "ms"  -- millisecond unit
+type instance ShowUnit (1 :% 1000000)    = "mcs" -- microsecond unit
+type instance ShowUnit (1 :% 1000000000) = "ns"  -- nanosecond unit
+
+type instance ShowUnit (60      :% 1) = "m"  -- minute unit
+type instance ShowUnit (360     :% 1) = "h"  -- hour unit
+type instance ShowUnit (86400   :% 1) = "d"  -- day unit
+type instance ShowUnit (604800  :% 1) = "w"  -- week unit
+type instance ShowUnit (1209600 :% 1) = "fn" -- fortnight unit
+
+instance KnownSymbol (ShowUnit unit) => Show (Time unit) where
+    show (Time rat) = let value = show (numerator rat) ++
+                           case denominator rat of
+                               1 -> ""
+                               n -> "/" ++ show n           in
+        value ++ symbolVal (Proxy @(ShowUnit unit))
```

Comment:
Alignment here is not looking really nice to me... Don't be afraid in introducing new variables! Personally, I would write it like this:

```haskell
instance KnownSymbol (ShowUnit unit) => Show (Time unit) where
    show (Time rat) = let numeratorStr   = show (numerator rat)
                          denominatorStr = case denominator rat of
                                                1 -> ""
                                                n -> '/' : show n
                      in numeratorStr ++ denominatorStr ++ symbolVal (Proxy @(ShowUnit unit))
```

But it's on your judgement :slightly_smiling_face: